README.md

```diff
@@ -516,29 +516,225 @@ func main() {
+	}
+
+	// Generate embeddings
+	var selectionsEmbeddings [][]float32
+	for _, selection := range selections {
+		embedding, err := getEmbedding(ctx, client, []string{selection})
+		if err != nil {
+			fmt.Printf("GetEmedding error: %v\n", err)
+			return
+		}
+		selectionsEmbeddings = append(selectionsEmbeddings, embedding)
+	}
+
+	// Write embeddings binary data to file
+	file, err := os.Create("embeddings.bin")
+	if err != nil {
+		fmt.Printf("Create file error: %v\n", err)
+		return
+	}
+	defer file.Close()
+
+	encoder := gob.NewEncoder(file)
+	err = encoder.Encode(selectionsEmbeddings)
+	if err != nil {
+		fmt.Printf("Encode error: %v\n", err)
+		return
+	}
+
+	return
+}
+```
+</details>
+
+<details>
+<summary>Embedding Similarity Search</summary>
+
+```go
+package main
+
+import (
+	"context"
+	"encoding/gob"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/sashabaranov/go-openai"
+)
+
+func getEmbedding(ctx context.Context, client *openai.Client, input []string) ([]float32, error) {
+	resp, err := client.CreateEmbeddings(ctx, openai.EmbeddingRequest{
+		Input: input, 
+		Model: openai.AdaEmbeddingV2,
+	})
+	
+	if err != nil {
+		return nil, err
+	}
+	
+	return resp.Data[0].Embedding, nil
+}
+
+// Sort the index in descending order of similarity
+func sortIndexes(scores []float32) []int {
+	indexes := make([]int, len(scores))
+	for i := range indexes {
+		indexes[i] = i
+	}
+	sort.SliceStable(indexes, func(i, j int) bool {
+		return scores[indexes[i]] > scores[indexes[j]]
+	})
+	return indexes
+}
+
+func main() {
+	ctx := context.Background()
+	client := openai.NewClient("your token")
+
+	//  "embeddings.bin" from exp: <Generate Embeddings>
+	file, err := os.Open("embeddings.bin")
+	if err != nil {
+		panic(err)
+	}
+	defer file.Close()
+
+	// load all embeddings from local binary file
+	var allEmbeddings [][]float32
+	decoder := gob.NewDecoder(file)
+	if err := decoder.Decode(&allEmbeddings); err != nil {
+		fmt.Printf("Decode error: %v\n", err)
+		return
+	}
+
+	// make some input you like
+	input := "I am a Golang Software Engineer, I like Go and OpenAI."
+
+	// get embedding of input
+	inputEmbd, err := getEmbedding(ctx, client, []string{input})
+	if err != nil {
+		fmt.Printf("GetEmedding error: %v\n", err)
+		return
+	}
+
+	// Calculate similarity through cosine matching algorithm
+	var questionScores []float32
+	for _, embed := range allEmbeddings {
+		// OpenAI embeddings are normalized to length 1, which means that:
+		// Cosine similarity can be computed slightly faster using just a dot product
+		score := openai.DotProduct(embed, inputEmbd)
+		questionScores = append(questionScores, score)
+	}
+
+	// Take the subscripts of the top few selections with the highest similarity
+	sortedIndexes := sortIndexes(questionScores)
+	sortedIndexes = sortedIndexes[:3] // Top 3
```

Comment:
Could we please add this section to the previous example and have one single example for embeddings?